From bf010e0ae57375774585069b484c962bef8a4013 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 21 Dec 2019 06:42:35 +0200
Subject: [PATCH] [SW-8362] make: pass proper SFLAGS to all assembly related targets

Add SFLAGS to lds reciepe
Add __ASSEMBLY__ definition to SFLAGS so general headers can be included
from assembly code and non relevant code can be ifdefed


Change-Id: I8314cf741ac1be28b0ff5b8e8068ce1c1939f705
---

diff --git a/Makefile b/Makefile
index 996d8f5..9e878c3 100644
--- a/Makefile
+++ b/Makefile
@@ -15,7 +15,7 @@
 endif
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
-SFLAGS := $(CFLAGS)
+SFLAGS := $(CFLAGS) -D__ASSEMBLY__
 
 PREBOOT_LD_SCRIPT := $(HL_DEV)/preboot.lds
 PPBOOT_LD_SCRIPT := $(HL_DEV)/ppboot.lds
@@ -64,7 +64,7 @@
 	@$(CC) -x c -E -P -nostdinc -o $@ $<
 
 %.lds: %.lds.S
-	@$(CC) -E -P -nostdinc -o $@ $<
+	@$(CC) $(SFLAGS) -E -P -nostdinc -o $@ $<
 
 %.o: %.c $(AUTO_GENERATED_FILE)
 	@$(CC) $(CFLAGS) -o $@ $<
